productivity-suite/piercing-library/src/message-bus/get-bus.ts

```diff
@@ -9,6 +10,8 @@ import { messageBusProp } from "./message-bus-prop";
  *                the message bus in the global scope is returned.
  * @returns       The most relevant message bus.
  */
+export function getBus(): MessageBus;
+export function getBus(element: Element): FragmentMessageBus;
```

Comment:
This overload means that we get access to the `onCleanup()` function only when we pass a ref.
This is not strictly asserted at runtime right now... E.g. if you pass a `ref` outside of a fragment you get a `GenericMessageBus` rather than a `FragmentMessageBus`. But we could add a check for that somehow.

Reply:
I guess we could add the `onCleanup()` method to the `GenericMessageBus` too but make it a noop?

Reply:
I guess... would there be a benefit in that?

Reply:
It would mean that we could avoid these overloads here, and also the user wouldn't need to care about whether it is inside a fragment or not...

Reply:
Yeah but I worry that it might break at runtime... if you do `getBus(ref).onCleanup(...)` outside of a fragment, then this will throw a null reference exception.

Reply:
ahh.... yeah good point...